container/container_unix.go

```diff
@@ -253,29 +253,40 @@ func (container *Container) IpcMounts() []Mount {
 	return mounts
 }
 
-// SecretMount returns the mount for the secret path
-func (container *Container) SecretMount() *Mount {
-	if len(container.SecretReferences) > 0 {
-		return &Mount{
-			Source:      container.SecretMountPath(),
-			Destination: containerSecretMountPath,
+// SecretMounts returns the mount for the secret path
+func (container *Container) SecretMounts() []Mount {
+	var mounts []Mount
+	for _, r := range container.SecretReferences {
+		// secrets are created in the SecretMountPath at a single level
+		// i.e. /var/run/secrets/foo
+		srcPath := filepath.Join(container.SecretMountPath(), filepath.Base(r.File.Name))
+		mounts = append(mounts, Mount{
+			Source:      srcPath,
+			Destination: r.File.Name,
 			Writable:    false,
-		}
+		})
 	}
 
-	return nil
+	return mounts
 }
 
 // UnmountSecrets unmounts the local tmpfs for secrets
 func (container *Container) UnmountSecrets() error {
-	if _, err := os.Stat(container.SecretMountPath()); err != nil {
-		if os.IsNotExist(err) {
-			return nil
+	for _, r := range container.SecretReferences {
+		srcPath := filepath.Join(container.SecretMountPath(), r.File.Name)
```

Comment:
In `SecretMounts` this uses the base name. The code here doesn't seem consistent.

Either way, we should break this out into a shared function.